docs/building.md

```diff
@@ -0,0 +1,167 @@
+
+```shell
+git clone https://github.com/fobnail/fobnail --recurse-submodules
+```
+
+Build process for Token firmware is by far the most complicated one. For that
+reason use of Fobnail SDK is strongly suggested. Follow [these instructions](/fobnail-sdk.md)
+to install it.
+
+With the SDK installed, one can prepare for the build process itself. Firmware
+can be build for a physical Fobnail Token or simulated on PC, with different
+preparatory steps required.
+
+#### Environment variables common for both targets
+
+Fobnail Token firmware is configured with environment variables passed to
+`build.sh`.
+
+Note that due to the way Docker mounts directories all files and directories
+pointed to by following variables must be located somewhere in `fobnail`
+directory.
+
+- `FOBNAIL_PO_ROOT` - **required** option, must point to valid PEM or DER file
+  with Platform Owner's root certificate. See [this document](/keys_and_certificates/#platform-owner-certificate-chain)
+  for description of PO certificate chain and instructions for building such.
+  For Fobnail Token firmware it is enough to create only [root CA certificate](https://fobnail.3mdeb.com/keys_and_certificates/#root-ca).
+  The rest of chain will be required for provisioning.
+
+- `FOBNAIL_EK_ROOT_DIR` - points to directory with TPM root certificates.
+  Fobnail repository includes [such directory](https://github.com/fobnail/fobnail/tree/main/tpm_ek_roots)
+  which normally should be used as `FOBNAIL_EK_ROOT_DIR`, but you may change it
+  if certificates supplied in repository became outdated. Either this or
+  `FOBNAIL_EXTRA_EK_ROOT` (or both) **must be specified**.
+
+- `FOBNAIL_EXTRA_EK_ROOT` - points to one specific TPM certificate. Useful when
+  [testing with TPM emulator](/tpm-simulators/), or to limit Fobnail usage to
+  one specific TPM vendor. Either this or `FOBNAIL_EK_ROOT_DIR` (or both)
+  **must be specified**.
+
+#### Building and flashing firmware to Fobnail Token
+
+For hardware setup instructions see [Flashing preparation](flashing_preparation.md).
+
+Building and flashing is performed by executing (from `fobnail` directory):
+
+```shell
+env FOBNAIL_PO_ROOT=root_ca.crt FOBNAIL_EK_ROOT_DIR=tpm_ek_roots \
+    ./build.sh -t nrf --run
+```
+
+A console with Fobnail Token output will be displayed. It is not required for
+normal operation but can be useful for debugging. It can be closed with Ctrl-C
+at any point. After that, Token can be used without nRF52840-DK - just plug it
+wherever it's needed.
+
+#### Building and running firmware locally
+
+Make sure you have the network set up properly. See [Networking setup](networking_setup.md)
+for instructions. For firmware to work properly you need the `fobnail0` network
+interface configured.
+
+To build and run firmware execute the following commands (from `fobnail`
+directory).
+
+```shell
+env FOBNAIL_LOG=info FOBNAIL_PO_ROOT=root_ca.crt \
+    FOBNAIL_EK_ROOT_DIR=tpm_ek_roots ./build.sh -t pc --run
```

Comment:
```
   Compiling pal_pc v0.1.0 (/build/pal/pal_pc)
    Finished dev [optimized + debuginfo] target(s) in 49.69s
     Running `target/x86_64-unknown-linux-gnu/debug/fobnail`
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 1, kind: PermissionDenied, message: "Operation not permitted" }', pal/pal_pc/src/net/mod.rs:26:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
```

Running:

```
env RUST_BACKTRACE=1 FOBNAIL_LOG=info FOBNAIL_PO_ROOT=root_ca.crt \
    FOBNAIL_EK_ROOT_DIR=tpm_ek_roots ./build.sh -t pc --run
```

does not change output

Reply:
@pietrushnic have you followed `networking_setup.md`?

Reply:
@krystian-hebel yeah, that is the problem.